demo-java/app/src/main/java/com/example/placesdemo/PlaceIsOpenActivity.java

```diff
@@ -0,0 +1,318 @@
+        }
+
+        Task<IsOpenResponse> placeTask = placesClient.isOpen(request);
+
+        placeTask.addOnSuccessListener(
+                (response) -> {
+                    textViewResponse.setText("Is place open? "
+                                                     + response.isOpen()
+                                                     + "\nExtra place details: \n"
+                                                     + StringUtil.stringify(place));
+                });
+
+        placeTask.addOnFailureListener(
+                (exception) -> {
+                    exception.printStackTrace();
+                    textViewResponse.setText(exception.getMessage());
+                });
+
+        placeTask.addOnCompleteListener(response -> setLoading(false));
+    }
+
+    /**
+     * Check if the place is open at the time specified in the input fields.
+     * Use the Place ID in the input field for the isOpenRequest.
+     */
+    @SuppressLint("SetTextI18n")
+    private void isOpenByPlaceId() {
+        clearViews();
+        dismissKeyboard(findViewById(R.id.editText_placeId));
+        setLoading(true);
+
+        IsOpenRequest request;
+
+        try {
+            request = IsOpenRequest.newInstance(getPlaceId(), isOpenCalendar.getTimeInMillis());
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            textViewResponse.setText(e.getMessage());
+            setLoading(false);
+            return;
+        }
+
+        Task<IsOpenResponse> placeTask = placesClient.isOpen(request);
+
+        placeTask.addOnSuccessListener(
+                (response) -> {
+                    textViewResponse.setText("Is place open? " + response.isOpen());
+                });
+
+        placeTask.addOnFailureListener(
+                (exception) -> {
+                    exception.printStackTrace();
+                    textViewResponse.setText(exception.getMessage());
+                });
+
+        placeTask.addOnCompleteListener(response -> setLoading(false));
+    }
+
+    //////////////////////////
+    // Helper methods below //
+    //////////////////////////
+
+    private void dismissKeyboard(EditText focusedEditText) {
+        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.hideSoftInputFromWindow(focusedEditText.getWindowToken(), 0);
+    }
+
+    private String getPlaceId() {
+        return ((TextView) findViewById(R.id.editText_placeId)).getText().toString();
+    }
+
+    /**
+     * Fetch the fields necessary for an isOpen request, unless user has checked the box to
+     * select a custom list of fields. Also fetches name and address for display text.
+     */
+    private List<Field> getPlaceFields() {
+        if (((CheckBox) findViewById(R.id.checkBox_useCustomFields)).isChecked()) {
+            return fieldSelector.getSelectedFields();
+        } else {
+            return new ArrayList<Field>(Arrays.asList(
```

Comment:
We do not need to explicitly point out the type here, the compiler catches it automatically.